Explorer/Assets/DCL/SDKComponents/MediaStream/Systems/VideoEventsSystem.cs

```diff
@@ -70,17 +74,29 @@ private void AppendMessage(in CRDTEntity sdkEntity, in MediaPlayerComponent medi
             );
         }
 
-        private static VideoState GetCurrentVideoState(IMediaControl mediaPlayerControl)
+        private static VideoState GetCurrentVideoState(IMediaControl mediaPlayerControl, double previousPlayingTimeCheck, float lastStateChangeTime)
         {
-            if (mediaPlayerControl.IsPlaying()) return VideoState.VsPlaying;
-            if (mediaPlayerControl.IsPaused()) return VideoState.VsPaused;
-            if (mediaPlayerControl.IsFinished()) return VideoState.VsNone;
-            if (mediaPlayerControl.IsBuffering()) return VideoState.VsBuffering;
-            if (mediaPlayerControl.IsSeeking()) return VideoState.VsSeeking;
+            // Important: while PLAYING or PAUSED, MediaPlayerControl may also be BUFFERING and/or SEEKING.
 
+            if (mediaPlayerControl.IsFinished()) return VideoState.VsNone;
             if (mediaPlayerControl.GetLastError() != ErrorCode.None) return VideoState.VsError;
+            if (mediaPlayerControl.IsPaused()) return VideoState.VsPaused;
+
+            VideoState state = VideoState.VsNone;
+            if (mediaPlayerControl.IsPlaying())
+            {
+                state = VideoState.VsPlaying;
+
+                if (mediaPlayerControl.GetCurrentTime().Equals(previousPlayingTimeCheck)) // Video is frozen
+                {
+                    state = mediaPlayerControl.IsSeeking() ? VideoState.VsSeeking : VideoState.VsBuffering;
 
-            return VideoState.VsNone;
+                    // If the seeking/buffering never ends, update state with error so the scene can react
+                    if ((Time.timeSinceLevelLoad - lastStateChangeTime) > MAX_VIDEO_FROZEN_SECONDS_BEFORE_ERROR)
```

Comment:
if you change it there, then also don't forget to change it here (but again, it is optional)

Reply:
tackled at https://github.com/decentraland/unity-explorer/pull/2474/commits/94162ae6cd1c59aa17ce5dbf1902129a7aa8405f